lib/Assertion.js

```diff
@@ -50,30 +50,22 @@ module.exports = class Assertions {
 
     evaluate(json) {
         const jsonValue = jsonpath.value(json, this.path);
-        if (this.operator === "==") {
+        if (this.operator === "==" || this.operator === "=~") {
             if (this.value === undefined) {
                 return jsonValue === undefined;
             }
 
+            let match = false;
             if (jsonValue !== undefined) {
-                return this.evaluateString(this.value, jsonValue);
-            }
-            return false;
-        } else if (this.operator === "=~") {
-            // If the user specifies the regex with forward-slashes to start and end, do special handling
-            let regexString = this.value;
-            let options = "";
-            if (this.value.startsWith("/")) {
-                regexString = regexString.substr(1);
-                // Now get the last /, and treat the part after as options
-                const endIndex = regexString.lastIndexOf("/");
-                if (endIndex + 1 < regexString.length) {
-                    options = regexString.substr(endIndex + 1);
+                if (Array.isArray(this.value)) {
+                    for (const value of this.value) {
+                        match = match || this.evaluateRegexOrString(this.operator, value, jsonValue)
```

Comment:
guess it should be a brea here because it is replacing match with the last occurrence